ultralytics/data/dataset.py

```diff
@@ -323,7 +323,9 @@ def get_labels(self):
                 if box[2] <= 0 or box[3] <= 0:
                     continue
 
-                cat_name = " ".join([img["caption"][t[0] : t[1]] for t in ann["tokens_positive"]])
+                caption = img["caption"]
+                tokens_positive = ann["tokens_positive"]
+                cat_name = " ".join([caption[t[0]: t[1]] for t in tokens_positive])
```

Comment:
@Lornatang How about let's keep the original `ann["tokens_positive"]` since the for loop would access it once no?

Reply:
Some json files may contain this field multiple times.
For example, YOLO_World official cc3m dataset json [cc3m_lite](https://huggingface.co/wondervictor/YOLO-World/blob/main/cc3m_pseudo_annotations.json).

Reply:
but here we are using `ann["tokens_positive"]` once, not like we are looping `img["caption"]` multiple times.
```python
cat_name = " ".join([img["caption"][t[0] : t[1]] for t in ann["tokens_positive"]])
```

Reply:
Ok, I see. This change may work if more GroundingDataset types are supported in the future, thanks.

Reply:
@Lornatang ok reverted the `ann["tokens_positive"]` in https://github.com/ultralytics/ultralytics/pull/18382/commits/052cb33a944f6caf04b9dc0abe3536fc1acaf49c